src/field.rs

```diff
@@ -224,45 +224,37 @@ pub mod tests {
     use blstrs::Scalar as Fr;
 
     use super::*;
-    use quickcheck::{Arbitrary, Gen};
+    use proptest::prelude::*;
 
     impl<F: LurkField> Arbitrary for FWrap<F> {
-        fn arbitrary(_: &mut Gen) -> Self {
-            let f = F::random(rand::thread_rng());
-            FWrap(f)
-        }
-    }
-
-    // For working around the orphan trait impl rule
-    #[derive(Clone, Debug, PartialEq, Eq)]
-    pub struct VecFWrap<F: LurkField>(pub Vec<F>);
+        type Parameters = ();
+        type Strategy = BoxedStrategy<Self>;
 
-    impl<F: LurkField> Arbitrary for VecFWrap<F> {
-        fn arbitrary(g: &mut Gen) -> Self {
-            let vec_f: Vec<FWrap<F>> = Arbitrary::arbitrary(g);
-            VecFWrap(vec_f.into_iter().map(|f| f.0).collect())
+        fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+            let strategy = Just(FWrap(F::random(rand::thread_rng())));
+            strategy.boxed()
         }
     }
 
-    fn repr_bytes_consistency<F: LurkField>(f1: FWrap<F>) -> bool {
+    fn repr_bytes_consistency<F: LurkField>(f1: FWrap<F>) {
         let bytes = f1.0.to_repr().as_ref().to_owned();
-        let f2 = <F as LurkField>::from_bytes(&bytes);
-        Some(f1.0) == f2
-    }
-
-    #[quickcheck]
-    fn prop_blstrs_repr_bytes_consistency(f1: FWrap<Fr>) -> bool {
-        repr_bytes_consistency(f1)
-    }
-
-    #[quickcheck]
-    fn prop_pallas_repr_bytes_consistency(f1: FWrap<pasta_curves::Fp>) -> bool {
-        repr_bytes_consistency(f1)
+        let f2 = <F as LurkField>::from_bytes(&bytes).expect("from_bytes");
+        assert_eq!(f1.0, f2)
     }
 
-    #[quickcheck]
-    fn prop_vesta_repr_bytes_consistency(f1: FWrap<pasta_curves::Fq>) -> bool {
+    proptest! {
+      #[test]
+      fn prop_bls_repr_bytes_consistency(f1 in any::<FWrap<Fr>>()) {
```

Comment:
Good idea. I tried using `test-case` for a minute to see if it would just work, but it seemed like it required some more thought, since we're parametrizing on the trait versus an expression. 

I think a simple macro would be fine if we start to have many of these kinds of tests. Or possibly we could even add an `Arbitrary` constraint on LurkField itself which would let us get rid of `FWrap` completely, but maybe that generates other orphan rule issues